index.js

```diff
@@ -143,42 +143,65 @@ module.exports = {
         return cb(new Error('addMembers only supports bendybutt-v1 feed IDs'))
       }
 
-      get(groupId, (err, { writeKey, root }) => {
+      get(groupId, (err, { root }) => {
         // prettier-ignore
         if (err) return cb(clarify(err, `Failed to get group details when adding members`))
 
         getRootFeedIdFromMsgId(root, (err, rootAuthorId) => {
           // prettier-ignore
           if (err) return cb(clarify(err, "couldn't get root id of author of root msg"))
 
-          const content = {
-            type: 'group/add-member',
-            version: 'v2',
-            groupKey: writeKey.key.toString('base64'),
-            root,
-            creator: rootAuthorId,
-            recps: [groupId, ...feedIds],
-          }
+          getTipEpochs(groupId, (err, tipEpochs) => {
+            // prettier-ignore
+            if (err) return cb(clarify(err, 'todo'))
 
-          if (opts.text) content.text = opts.text
+            const getFeed = opts?._feedKeys
+              ? (cb) => cb(null, { keys: opts._feedKeys })
+              : findOrCreateAdditionsFeed
 
-          const getFeed = opts?._feedKeys
-            ? (cb) => cb(null, { keys: opts._feedKeys })
-            : findOrCreateAdditionsFeed
+            getFeed((err, additionsFeed) => {
+              // prettier-ignore
+              if (err) return cb(clarify(err, 'Failed to find or create additions feed when adding members'))
 
-          getFeed((err, additionsFeed) => {
-            // prettier-ignore
-            if (err) return cb(clarify(err, 'Failed to find or create additions feed when adding members'))
+              const options = {
+                isValid: isAddMember,
+                tangles: ['members'],
+                feedKeys: additionsFeed.keys,
+              }
+              pull(
+                pull.values(tipEpochs),
+                pull.asyncMap((tipEpoch, cb) => {
+                  getPredecessorEpochs(
+                    groupId,
+                    tipEpoch.id,
+                    (err, predecessors) => {
+                      // prettier-ignore
+                      if (err) return cb(clarify(err, 'todo'))
+
+                      const content = {
+                        type: 'group/add-member',
+                        version: 'v2',
+                        secret: tipEpoch.secret.toString('base64'),
+                        oldSecrets: predecessors.map((pred) =>
+                          pred.secret.toString('base64')
+                        ),
```

Comment:
just realized we're sending this also on re-additions after an exclusion. we should only send it when adding a new member. add exception.